airflow/utils/state.py

```diff
@@ -46,6 +46,9 @@ class TaskInstanceState(str, Enum):
     REMOVED = "removed"  # Task vanished from DAG before it ran
     SCHEDULED = "scheduled"  # Task should run and will be handed to executor soon
 
+    # Set by executor
+    ADOPTED = "adopted"
+
```

Comment:
KubernetesExecutor is a part of the `cncf.kuberntes` provider, which could be used with the old version of Airlfow (the current min airflow version for providers is 2.6.0), which means that this state could be unavailable for the users who use the latest version of k8s provider with an old version of Airflow.

Also, the TI will never have this state in the metadata, so maybe you can find another better way to tell the executor to free a slot.

Reply:
It make sense. Initially I though of same thing. Here is what I can think of keep that adopted to Kubernetes executor.